### ISSUE TITLE
pkgdown layout is broken

### ISSUE THREAD

**Issue by pawelru** [insightsengineering/teal#1359]
Visible only for `main`: https://insightsengineering.github.io/teal/main/

![image](https://github.com/user-attachments/assets/d7446cab-74f8-456c-ac48-813630f797d9)

might be related: https://github.com/insightsengineering/r-pkgdown-multiversion/issues/22

**Comment by pawelru** [comment]
Hey @cicdguy is this something close to what you have just fixed?

**Comment by pawelru** [comment]
I think it is related. I run the `pkgdown::build_site()` locally and it render nicely. That means that our injection is not working correctly.

Have a look at this https://github.com/insightsengineering/teal/commit/9745ffa9fb5badc3e2d780924c0212970e4c3bd9#diff-539e48ef777e8ae44686d203f7a8d44099150953e61b552694032fe7a3a428ba. This is the last commit on main from 5h ago so it includes yesterday fix.
We did the modifications in line 79 whereas it should have been 78

```
<ul class="navbar-nav me-auto">

  (...)

  <!-- changelog -->
  <li class="nav-item">
    <a class="nav-link" ...> Changelog</a>
  </li>

                                                        # <- this is where it's should be injected

  <!-- Reports -->
  <li class="nav-item dropdown">                        # line 78 in 9745ffa9fb5badc3e2d780924c0212970e4c3bd9 that I refer to
                                                        # <- this is where it's being injected right now
    <button ...>Reports</button>
    <ul ...>
      <li>...</li>
      <li>...</li>
      ...
    </ul>
  </li>

</ul>
```

In other words: line 78 belongs to Reports and we should be right before it.

On main we have:

<img width="583" alt="image" src="https://github.com/user-attachments/assets/6ace7db4-8189-4d54-9e4f-d39beccb4b39">

That comment should be inside `<ul>` (next to all the `<li>` tags) and not inside `<li>`.

WDYT? @cicdguy

**Comment by pawelru** [comment]
1. If I were to rewrite this functionality I would do this as a _pre-processing_ before the main pkgdown job of creating the docs. That step would modify the `_pkgdown.yaml` and add hrefs there. Then we can run the pkgdown as usual.
This is in the opposite of _post-processing_ after the pkgdown run. What we are doing is to modify what pkgdown produces. This way we will be always affected by possible breaking changes in pkgdown.
Either way - it's a considerable amount of work so let's try to find a quick win here.
2. I think we can add the pkgdown version in the footer but I don't really follow how this could help. We are injecting in the navbar element. How edits in footer could help? Apologies for my weak understanding of the logic.

**Comment by pawelru** [comment]
Hey @cicdguy did you have a chance to look into this?

**Comment by pawelru** [comment]
Awesome! Thank you!

**Comment by pawelru** [comment]
This got resolved. Thanks again @cicdguy
